JakartaEE10対応で、JavaEE由来の表現を修正

tisuchida opened the pull request:
各ドキュメントに、ドキュメント内でのJakartaEE仕様名への読み替えを案内する項目を追加。
またUI標準(画面)については、JSPについての説明をJakartaEE前提のシステムでの記載に改めた。

ドキュメント毎の対応有無は以下の通り（対応が必要なものにはチェックを入れている。不要なものは不要と判断した理由を付記した）。
- 開発プロセス標準
  - [X] Nablarchを使用したシステム開発における標準WBS.xlsx
- アプリケーション開発標準
  - 設計標準
    - [ ] DB設計標準.docx
      - データベースの設計についての記載のみで、Javaに関する言及が出て来ない内容のため。
    - [X] UI標準(画面).xlsx
    - [ ] UI標準(画面)別冊_UI部品カタログ.xlsx
      - UI標準(画面)の別冊のため、本体側で読み替えを案内すれば十分なため。
    - [ ] UI標準(帳票).xlsx
      - 帳票の見た目についての記載のみで、Javaに関する言及が出て来ない内容のため。
    - [ ] 共通コンポーネント設計標準.docx
      - 共通コンポーネントの責務、種別についての記載のみで、Javaに関する言及が出て来ない内容のため。
  - テスト標準
    - [X] 単体テスト標準.xlsx
- 設計ドキュメント
  - [X] Nablarch開発標準設計書一覧.xlsx
  - [ ] フォーマット
    - 実装に関する情報は設計書には記載しない前提であり、Javaに関する言及が出て来ない内容のため。
  - [ ] サンプル
    - 実装に関する情報は設計書には記載しない前提であり、Javaに関する言及が出て来ない内容のため。
- en/Application_Development_Standards
  - Design_Standards
    - [ ] Common_Component_Design_Standards.docx
      - 共通コンポーネント設計標準と同じ理由で不要とした。
    - [ ] DB_Design_Standards.docx
      - DB設計標準と同じ理由で不要とした。
  - Test_Standards
    - [X] Unit_Test_Standards.xlsx
- [X] README.md

tisuchida committed:
標準WBSの記載を変更

- 1.4. Nablarch6を使用する場合の記載の読み替えについて を追加
- 2. WBS K17,G28 JSP自動生成ツールの記載を削除

tisuchida committed:
フィードバックを踏まえて標準WBSの記載変更

tisuchida committed:
UI標準(画面)の記載修正

- 1.3. 本書におけるJakartaEE仕様への読み替えについて を追加
- 2.2.1. 仕様技術 JSPの記載をJakarta Server Pages 3.1向けのものに修正

tisuchida committed:
単体テスト標準の記載修正

    - 1.3. 本書におけるJakartaEE仕様への読み替えについて を追加

tisuchida committed:
Nablarch開発標準設計書一覧の記載修正

- 1.2. 本書におけるJakartaEE仕様への読み替えについて を追加

tisuchida committed:
Unit Test Standardsの記載修正

- 1.3. The reading into the JakartaEE specification in this document を追加

kumagoro1202 commented:
Excelファイルは全シートカーソルをA1セル、倍率100%にそろえてください。

kumagoro1202 commented:
- Nablarchを使用したシステム開発における標準WBS

WBSシートK17セルから「PG・UTI工程において、JSP自動生成ツールを使用することで、HTML→JSP変換の作業負荷を軽減できる。」という文言を消した後に「画面イメージはHTMLで作成すること。」だけを残しているが、JSP自動生成ツールのくだりがないとHTMLで作成する意味がない。HTMLで作成することという文言だけを残す必要性がない。削除するか別の理由、例えば「HTMLで表現できない画面を作成してしまわないように」とかの理由付けが必要。

kumagoro1202 commented:
- UI標準(画面)

変更履歴に記載がないのは意図したものですか？プロジェクトではこれをもとに作るもので変更履歴はプロジェクトが作成したところから記載するものという考え？

kumagoro1202 commented:
- UI標準(画面)

2シートF54セル、説明にJakarta Server Pagesの内容を記載するため修正しているのであればこのセルの記載もJakarta Server Pagesに修正するべき。あと説明欄にある「Jakarta Server Pagesの略。」は誤り。あくまでJSPという表記がJakarta Server Pagesのことを指しているだけで、Jakarta Server Pagesの略としてJSPを使うことは許可されていない。
https://jakarta.ee/legal/acronym_guidelines/

kumagoro1202 commented:
-  UI標準(画面)別冊_UI部品カタログ.xlsx
   - UI標準(画面)の別冊のため、本体側で読み替えを案内すれば十分なため。

[ガイドライン](https://jakarta.ee/legal/acronym_guidelines/)を見ると

> A reader should not have to scroll, turn a page, or click a link to get to the sentence.

とあるので、本体側で十分とは判断できないと思います。別冊にも読み替えの案内を入れた方が良いのではないでしょうか。

kumagoro1202 commented:
- 単体テスト標準

これに読み替えの案内をつけたのはなぜ？このドキュメントにJakarta EEの仕様が出てくるようには見えないし、現在も出てない。どういう基準で追加している？

kumagoro1202 commented:
設計ドキュメント

- フォーマット
   - 内容はすべてプロジェクト側で作成するため。

どういう意味？プロジェクトが作成することと修正が不要なことの関係がわからない。

- サンプル
   - サンプルを修正して成果物にすることはないため。

これもサンプル修正して成果物にしないことと修正が不要なことの関係がわからない。

tisuchida commented:
https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177366382

記載を削除しました。

tisuchida commented:
https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177400778

記載を修正しました。

tisuchida commented:
https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177419145

読み替えの案内を追加しました。

hijiri-tis commented:
CHANGELOGには今回の修正内容を書きましょう

tisuchida committed:
標準WBSのレビュー指摘修正実施

- WBSシートK17の補足事項の記載削除
  - https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177366382

tisuchida commented:
https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177366382

補足事項の記載を削除しました。
7aa23e07c381ab400f0518081574c4882a3be588

tisuchida commented:
https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177376401

ご認識の通り、変更履歴はプロジェクトで作成した時点から記載される想定のため、今回の修正内容は変更履歴に反映していません。
（UI標準に対する過去の変更も記載されていないので、そのようなポリシーと理解しています）

tisuchida committed:
UI標準（画面）のレビュー指摘対応

- 2シートF54セル JSPの説明ではなくJakarta Server Pagesの説明に修正
- 不適切な表現の削除
  - https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177400778

tisuchida commented:
https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177400778

Jakarta Server Pagesに対する説明になるよう修正しました。
432082b36b67dff6afc343fe25e368e06b78d9ba

tisuchida committed:
UI標準（画面）別冊_UI部品カタログのレビュー指摘修正

- 読み替えの案内を追加
  - https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177419145

tisuchida commented:
https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177419145

読み替えの案内を追加しました。
dabad992ce9a185ca98a4d58e8234af14306cb15

tisuchida committed:
JBatch表記をjBatch表記に修正、設計書一覧に変更履歴追加

tisuchida commented:
JBatch表記をjBatch表記に改めました。
また、設計書一覧の変更履歴に今回の変更内容の記載が漏れていたため、合わせて記載しました。

dc787d4bec47d2fbd0f718d17e3972bdf8d50b88

tisuchida committed:
Revert "単体テスト標準の記載修正"

This reverts commit fd35bb173c5fca5990a5d7df98a9066baab153a8.

tisuchida committed:
Revert "Unit Test Standardsの記載修正"

This reverts commit bb35183755172f1c36d9eced65aa961be970c390.

tisuchida committed:
Unit_Tests_Standardsのレビュー指摘修正

- 4シートF28からの注釈
  日本語版ドキュメントと比較し、不要となっている記載の削除
  - https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177768185

tisuchida commented:
https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177768185

当初は、英語版の「Unit_Tests_Standards.xlsx」で「JSP」が引っかかったため、対応する日本語版もプロジェクトのカスタマイズにより「JSP」等の文言が記載される可能性があると考え、追記しました。

ただ、英語版の該当箇所付近（英語版のシート4、F28付近）には日本語版にはない以下の記載があります。
```
*1 Generally, checks need to be performed for all devices and browsers defined in the UI standards, but a check in one browser is sufficient if the following premises are used, as the widgets and JavaScript components provided by the UI development platform guarantees these points in these cases.		
	・The layouts of the widgets and JavaScript components have been verified for all relevant devices and browsers.	
	・Individual styles are not defined or applied in the business screen JSP.	
	・Individual JavaScript items are not defined or used in the business screen JSP.	

* Sanitizing: This unit test case is not required as implementations that may cause sanitizing omissions are prohibited in JSP coding convention. (Use the JSP static analysis tool to check that there are no implementations that violate this convention.)		
```

記載内容に差異がある経緯は不明ですが、以下の理由により不要と考え、日本語版に合わせて記載を削除しました。
- 上記の記載でUI開発基盤に言及している箇所はNablarch6では不要となる（UI開発基盤が削除されるため）
- JSPコーディング規約でサニタイズ漏れとなる実装を禁止しているので単体テスト不要、といった記載は（JSP静的解析ツールでチェックできるとは言え）必ずしも真とは言えない

上記によりJSPの文言が英語版からも消えます。それを踏まえ、読み替えの案内も削除（当該コミットのrevert）しました。

(revert コミット)
1ee74889c9c0c5367cbdd354812a88b77b045033
02c7ab53678aadef02ad272d3f725b96477791ed
(記載の削除コミット)
9f979e95567f4167520221b415c1d40b9708be9b

tisuchida committed:
JSR352表記をJakarta Batch表記に修正

tisuchida commented:
README.mdにあったJSR352の表記をJakarta Batchに変更しました。
（検知はしていたものの、対応が漏れていたものです。申し訳ありません）

5179af10e770f1155ccadeae77af4838a6c6b45d

tisuchida commented:
https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177774917

記載があいまいでした。
[設計書には細かい実装に関する情報は記載しない](https://github.com/nablarch-development-standards/nablarch-development-standards/blob/master/README.md#%E8%A8%AD%E8%A8%88%E6%9B%B8%E3%81%AB%E8%A8%98%E8%BC%89%E3%81%99%E3%81%B9%E3%81%8D%E6%83%85%E5%A0%B1%E3%82%92%E6%A5%B5%E5%8A%9B%E5%B0%91%E3%81%AA%E3%81%8F%E3%81%97%E9%96%8B%E7%99%BA%E7%94%9F%E7%94%A3%E6%80%A7%E3%82%92%E5%90%91%E4%B8%8A%E3%81%95%E3%81%9B%E3%81%BE%E3%81%99)想定であり、Javaに関連した文言は出て来ない内容のため、が理由となります。

descriptionを修正しました。

tisuchida committed:
CHANGELOGに今回の変更を記載

- CHANGELOGの日付を2024/09/30に設定
- 他ドキュメントの変更履歴も併せて2024/09/30に設定

tisuchida commented:
https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2178566729

CHANGELOGに今回変更内容を記載しました。
合わせて、変更履歴を更新した他ドキュメントの日付をCHANGELOGに合わせました。
5f69e77193f00e99fb1c991b24b4c8d0147c2444

tisuchida commented:
https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177361778

いずれもA1カーソル合わせ、倍率100%に設定しました。

kumagoro1202 commented:
> [#39 (comment)](https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177376401)
> 
> ご認識の通り、変更履歴はプロジェクトで作成した時点から記載される想定のため、今回の修正内容は変更履歴に反映していません。 （UI標準に対する過去の変更も記載されていないので、そのようなポリシーと理解しています）

変更履歴に載せなくても変更日付だけは変えておいてください。いつ手が入ったかわかりやすいので。

kumagoro1202 commented:
> * JSPコーディング規約でサニタイズ漏れとなる実装を禁止しているので単体テスト不要、といった記載は（JSP静的解析ツールでチェックできるとは言え）必ずしも真とは言えない

コーディング規約で禁止しているからテスト不要ではなく、JSP静的解析ツールでチェックして検知できるからテスト不要という文章です。該当の記載を削除するのは良いですが、文章の意味は正しく読解してください。

tisuchida committed:
変更履歴中の変更日付を、実際の更新日に修正

tisuchida committed:
CHANGELOGで、今回分の変更が2.3版となるよう修正

tisuchida commented:
https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2182070879
今回修正していないドキュメントも含め、変更日付を最終更新日に変更しました。

e905039d71a938bbba68393b46781a3a3dd8f340

tisuchida commented:
改めて変更履歴を見直したところ、2.2版が最新版であったため、今回の変更内容は2.3版の箇所に追記するように修正しました。


4112175a8909961b9a450a09711bfca7effb442f

tisuchida committed:
日本語と不整合になっている箇所の削除

- 4.1.1. Viewpoints 表の番号11~14を削除
- 4.1.2. Details of viewpoints (2)を削除

kumagoro1202 commented:
> [#39 (comment)](https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2177768185)
> 
> 当初は、英語版の「Unit_Tests_Standards.xlsx」で「JSP」が引っかかったため、対応する日本語版もプロジェクトのカスタマイズにより「JSP」等の文言が記載される可能性があると考え、追記しました。
> 
> ただ、英語版の該当箇所付近（英語版のシート4、F28付近）には日本語版にはない以下の記載があります。
> 
> ```
> *1 Generally, checks need to be performed for all devices and browsers defined in the UI standards, but a check in one browser is sufficient if the following premises are used, as the widgets and JavaScript components provided by the UI development platform guarantees these points in these cases.		
> 	・The layouts of the widgets and JavaScript components have been verified for all relevant devices and browsers.	
> 	・Individual styles are not defined or applied in the business screen JSP.	
> 	・Individual JavaScript items are not defined or used in the business screen JSP.	
> 
> * Sanitizing: This unit test case is not required as implementations that may cause sanitizing omissions are prohibited in JSP coding convention. (Use the JSP static analysis tool to check that there are no implementations that violate this convention.)		
> ```
> 
> 記載内容に差異がある経緯は不明ですが、以下の理由により不要と考え、日本語版に合わせて記載を削除しました。
> 
> * 上記の記載でUI開発基盤に言及している箇所はNablarch6では不要となる（UI開発基盤が削除されるため）
> * JSPコーディング規約でサニタイズ漏れとなる実装を禁止しているので単体テスト不要、といった記載は（JSP静的解析ツールでチェックできるとは言え）必ずしも真とは言えない
> 
> 上記によりJSPの文言が英語版からも消えます。それを踏まえ、読み替えの案内も削除（当該コミットのrevert）しました。
> 
> (revert コミット) [1ee7488](https://github.com/nablarch-development-standards/nablarch-development-standards/commit/1ee74889c9c0c5367cbdd354812a88b77b045033) [02c7ab5](https://github.com/nablarch-development-standards/nablarch-development-standards/commit/02c7ab53678aadef02ad272d3f725b96477791ed) (記載の削除コミット) [9f979e9](https://github.com/nablarch-development-standards/nablarch-development-standards/commit/9f979e95567f4167520221b415c1d40b9708be9b)

口頭で伝えましたが、英語版と日本語版で注記だけでなく乖離があります。英語版が修正漏れていると思うので確認してください。

tisuchida committed:
欠けていた目次項目（1.2）を追加

tisuchida commented:
https://github.com/nablarch-development-standards/nablarch-development-standards/pull/39#issuecomment-2182265115

全体的に見直し、項目レベルで乖離がある箇所を、日本語版ドキュメントに合わせて修正しました。
また日本語の目次項目に欠けがあったので、合わせて修正しました。

9fdbd2768de5a1253c97283a3dbab41b2849a9ae
263e020b2815f7899c905b24ebfcf89bca6129c7

kumagoro1202 commented:
https://github.com/nablarch-development-standards/nablarch-development-standards/tree/jakartaee/fix-javaee-expressions/020_%E3%82%A2%E3%83%97%E3%83%AA%E3%82%B1%E3%83%BC%E3%82%B7%E3%83%A7%E3%83%B3%E9%96%8B%E7%99%BA%E6%A8%99%E6%BA%96

スタイルガイドのリンクが古いので、あわせて修正してください。
https://github.com/Fintan-contents/coding-standards

tisuchida committed:
Revert "欠けていた目次項目（1.2）を追加"

This reverts commit 263e020b2815f7899c905b24ebfcf89bca6129c7.

tisuchida committed:
Revert "変更履歴中の変更日付を、実際の更新日に修正"

This reverts commit e905039d71a938bbba68393b46781a3a3dd8f340.

tisuchida committed:
変更履歴中の変更日付を、実際の更新日に修正

合わせて、単体テスト標準の目次項目の欠け（1.2）を修正

tisuchida committed:
スタイルガイドのリンクをFintanコンテンツのリンクに修正

hijiri-tis commented on 020_アプリケーション開発標準/020_テスト標準/単体テスト標準.xlsx:
以前に変更履歴を削除した対応を今回のリリースで反映すると思うので、ファイルの作成日も他と合わせて今回リリース日にしておくのが妥当なように思いました。（英語版も同じかな）

hijiri-tis replied:
同じ理由から、サンプルにあるドメイン定義書もかな。
サンプルなのでまぁいいかもだけど、前回修正時に日付を修正していそうであれば、合わせて今回のリリース日にしておいた方が後でどこの日付使ってるんだってならないからいいと思いました。

以前の対応はおそらくこれ https://github.com/nablarch-development-standards/nablarch-development-standards/pull/38

tisuchida replied:
単体テスト標準・サンプルのドメイン定義書について日付修正しました。
英語版の単体テスト標準は修正済みのため、特に変更していません。
dd9ff0d7c2c6d94b3836a102715ed6470f911556

tisuchida committed:
変更日付を次回リリース予定日に変更
